k2/csrc/array.h

```diff
@@ -103,19 +160,49 @@ struct Array3 {
                            // .. indexes2[indexes1[size]-1] should be defined.
 
   Ptr data;  // `data` might be an actual pointer, or might be some object
-             // supporting operator [].  data[indexes[0]] through
-             // data[indexes[size] - 1] must be accessible through this
+             // supporting operator [].  data[indexes2[indexes1[0]]] through
+             // data[index2[indexes1[size] - 1]] must be accessible through this
              // object.
 
-  Array2<I, Ptr> operator[](I i) {
-    // TODO(haowen): fill real data here
-    Array2<I, Ptr> array;
+  Array2<Ptr, I> operator[](I i) {
```

Comment:
why do we need a non-const version as we don't change any value of `array3` in this function

Reply:
Removed.